### ISSUE TITLE
Pad of Broadcast IterDomain broadcasts into the pad region

### ISSUE THREAD

**Issue by jacobhinkle** [NVIDIA/Fuser#868]
PR #841 enabled us to `cat` along broadcast `IterDomain`s, by allowing those pads. In that case, the pad values are unused since they are predicated away. If we actually inspect the pad values, it seems they are incorrect: the broadcast value is being used in place of the pad value. See below:

```python
import torch
import torch.nn.functional as F
from nvfuser import FusionDefinition, DataType

torch.manual_seed(0)
inputs = [
        torch.randn((1,), device='cuda'),
        ]

with FusionDefinition() as fd:
    x = fd.from_pytorch(inputs[0])
    x_pad = fd.ops.pad(x, [1, 1])
    fd.add_output(x_pad)

nvf_out = fd.execute(inputs)
# With NVFUSER_DUMP=fusion_ir_preseg:
# Inputs:
#   T0_g[ bS0{1} ], float
# Outputs:
#   T1_g[ iS2{( ( ( (nvfuser_index_t)(1) ) + 1 ) + ( (nvfuser_index_t)(1) ) )}rf ], float
# 
# %kernel_math {
# i6 = (nvfuser_index_t)(1);
# i8 = (nvfuser_index_t)(1);
# T1_g[ iS2{( ( ( (nvfuser_index_t)(1) ) + 1 ) + ( (nvfuser_index_t)(1) ) )}rf ]
#    = pad( T0_g[ bS0{1} ], {i6, i8} )
# }

print(nvf_out[0])
# tensor([-0.9247, -0.9247, -0.9247], device='cuda:0')
# Should be tensor([0, -0.9247, 0], device='cuda:0')

assert(torch.equal(nvf_out[0], F.pad(inputs[0], (1, 1))))
```
We see that the output `IterType` is `Iteration`, which rules out the possibility that the output was interpreted as an expanded broadcast instead of a pad.

Related to #21

**Comment by jacobhinkle** [comment]
This is where the padding predicate is built: https://github.com/NVIDIA/Fuser/blob/main/csrc/device_lower/pass/index.cpp#L1480-L1490. The problem is that `producer_idx` is constant zero since the producer is a broadcast. We need some way to force this to treat a broadcast as a size-1 `Iteration` domain in this circumstance at `getProducerAllocationIndices`: https://github.com/NVIDIA/Fuser/blob/main/csrc/index_compute.cpp#L1974-L1979.

**Comment by jacobhinkle** [comment]
Special casing for `producer_root_id->getIterType() == IterType::Broadcast` in `IndexLowering::handle(PadOp*)` is probably the clearest way to go here. I will PR it soon.
